test/fakes/fakedatastore/fakedatastore.go

```diff
@@ -7,38 +7,34 @@ import (
 	"sync/atomic"
 	"testing"
 
+	"github.com/sirupsen/logrus/hooks/test"
 	"github.com/stretchr/testify/require"
 
 	"github.com/spiffe/spire/pkg/common/util"
 	"github.com/spiffe/spire/pkg/server/plugin/datastore"
 	"github.com/spiffe/spire/pkg/server/plugin/datastore/sql"
-	spi "github.com/spiffe/spire/proto/spire/common/plugin"
-	"github.com/spiffe/spire/test/spiretest"
 )
 
 var (
 	nextID uint32
 )
 
 type DataStore struct {
-	datastore.UnsafeDataStoreServer
-
 	ds   datastore.DataStore
 	errs []error
 }
 
 var _ datastore.DataStore = (*DataStore)(nil)
 
 func New(tb testing.TB) *DataStore {
-	var ds datastore.Plugin
-	spiretest.LoadPlugin(tb, sql.BuiltIn(), &ds)
-
-	_, err := ds.Configure(context.Background(), &spi.ConfigureRequest{
-		Configuration: fmt.Sprintf(`
-			database_type = "sqlite3"
-			connection_string = "file:memdb%d?mode=memory&cache=shared"
-		`, atomic.AddUint32(&nextID, 1)),
-	})
+	log, _ := test.NewNullLogger()
```

Comment:
I can not think in a good use for it, but may we allow a way to setup a fake with a hook for logs?  
this fake is special, and we may need access to some of thoe logs,
at the same time it may be an overkill because we dont want to test "sql" implementation but results..
but we can create some tests with end to end logs in some scenaries,
What do you think about it?

Reply:
[YAGNI](https://martinfowler.com/bliki/Yagni.html) :) Let's add that if/when it ever becomes convenient. I don't imagine it will though. Like you said, hard to imagine wanting the DS logs from unit-tests outside of the DS.